frontend/src/components/App/Home/index.tsx

```diff
@@ -160,30 +161,106 @@ function ClusterStatus({ error }: { error?: ApiError | null }) {
 
 export default function Home() {
   const history = useHistory();
-  const clusters = useClustersConf() || {};
+  const clustersConf = useClustersConf() || {};
+  const clusters = useMemo(() => {
+    return clustersConf;
+  }, [clustersConf]);
 
   if (!helpers.isElectron() && Object.keys(clusters).length === 1) {
     history.push(createRouteURL('cluster', { cluster: Object.keys(clusters)[0] }));
     return null;
   }
-
   return <HomeComponent clusters={clusters} key={Object.keys(clusters).join('')} />;
 }
 
 interface HomeComponentProps {
   clusters: { [name: string]: Cluster };
 }
 
+/**
+ * Custom hook that caches the result and only returns a new result if there is a deep equality change.
+ *
+ * @template T - The type of the value to be memoized.
+ * @param factory - The function that returns the value to be memoized.
+ * @param deps - The dependencies array.
+ * @returns The memoized value.
+ *
+ * @example
+ * const memoizedValue = useDeepCompareMemo(computeExpensiveValue(a, b), [a, b]);
+ */
+function useDeepCompareMemo<T>(value: T, deps: any[]): T {
+  const previousDepsRef = useRef<any[]>([]);
+  const previousValueRef = useRef<T>();
+
+  if (!isEqual(previousDepsRef.current, deps)) {
+    previousDepsRef.current = deps;
+    previousValueRef.current = value;
+  }
+
+  return previousValueRef.current as T;
+}
+
 function HomeComponent(props: HomeComponentProps) {
   const { clusters } = props;
-  const customNameClusters = Object.values(clusters).map(c => ({
-    ...c,
-    name: c.meta_data?.extensions?.headlamp_info?.customName || c.name,
-  }));
+  const customNameClusters = useMemo(() => {
+    return Object.values(clusters).map(c => ({
+      ...c,
+      name: c.meta_data?.extensions?.headlamp_info?.customName || c.name,
+    }));
+  }, [clusters]);
+
+  const customClusterNames = useMemo(() => {
+    return customNameClusters.map(c => c.name);
+  }, [customNameClusters]);
+
+  const clusterList = useMemo(() => {
+    return Object.values(clusters);
+  }, [clusters]);
+
+  const versionsErrors = useClustersVersion(clusterList);
+  const [versions, errors] = useDeepCompareMemo(versionsErrors, [versionsErrors]);
+  const warningsList = Event.useWarningList(customClusterNames);
+  const warningsMap = useDeepCompareMemo(warningsList, [warningsList]);
+
+  console.log('versions', versions);
+  console.log('clusterList', clusterList);
+  console.log('warningsList', warningsList);
+  console.log('customClusterNames.join', customClusterNames.join('-'));
+
+  const rendered = useMemo(() => {
+    return (
+      <HomePure
+        customNameClusters={customNameClusters}
+        customClusterNames={customClusterNames}
+        versions={versions}
+        errors={errors}
+        warningsMap={warningsMap}
+      />
+    );
+  }, [customClusterNames, customClusterNames, versions, errors, warningsMap]);
```

Comment:
I don't understand how the memo helps here. The component shouldn't refresh anyway unless one of its state or props changes, and AFAICS, the properties are now all extracted and being passed to them but also added here as deps, so this is having the same effect as if we just pass the properties.